modin/backends/pandas/query_compiler.py

```diff
@@ -1180,6 +1180,56 @@ def sort_index(self, **kwargs):
         )
         return self.__constructor__(new_modin_frame)
 
+    def melt(self, *args, **kwargs):
+        def _convert_to_list(x):
+            if is_list_like(x):
+                x = [*x]
+            elif x is not None:
+                x = [x]
+            else:
+                x = []
+            return x
+
+        unique_name = "__-index-__"
+
+        # inserting column to sort in a correct order futhure
+        prepaired = self.insert(
+            loc=len(self.columns), column=unique_name, value=np.arange(len(self.index))
```

Comment:
This is going to be slow because you are using a `numpy` array, which is not distributed.

Reply:
Is there a way not to use this?

Reply:
`melt` reimplemented